test/object-store/sync/flx_sync.cpp

```diff
@@ -1323,6 +1322,33 @@ TEST_CASE("flx: asymmetric sync", "[sync][flx][app]") {
     }
 }
 
+TEST_CASE("flx: asymmetric sync with embedded objects") {
+    FLXSyncTestHarness::ServerSchema server_schema;
+    // server_schema.queryable_fields = {"queryable_str_field"};
+    server_schema.schema = {
+        {"Asymmetric",
+         ObjectSchema::IsAsymmetric{true},
+         {
+             {"_id", PropertyType::ObjectId, Property::IsPrimary{true}},
+             {"embedded_obj", PropertyType::Object | PropertyType::Nullable, "Asymmetric_embedded_obj"},
+         }},
+        {"Asymmetric_embedded_obj",
+         ObjectSchema::IsEmbedded{true},
+         {
+             {"value", PropertyType::String | PropertyType::Nullable},
+         }},
+    };
+
+    FLXSyncTestHarness harness("asymmetric_sync", server_schema);
+
+    harness.load_initial_data([&](SharedRealm realm) {
+        CppContext c(realm);
+        Object::create(
+            c, realm, "Asymmetric",
+            util::Any(AnyDict{{"_id", ObjectId::gen()}, {"embedded_obj", AnyDict{{"value", std::string{"foo"}}}}}));
```

Comment:
I know this will exercise the create/set link codepaths, but can we add a test here that updates the embedded field to null and then to a new embedded field as well?

Reply:
sure. I'll add that